src/main/java/io/confluent/idesidecar/restapi/cache/AdminClients.java

```diff
@@ -0,0 +1,105 @@
+import io.confluent.idesidecar.restapi.connections.ConnectionState;
+import io.confluent.idesidecar.restapi.events.Lifecycle;
+import io.confluent.idesidecar.restapi.kafkarest.exceptions.AdminClientInstantiationException;
+import jakarta.annotation.Nullable;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.event.ObservesAsync;
+import jakarta.inject.Inject;
+import java.util.Map;
+import java.util.Properties;
+import org.apache.kafka.clients.admin.AdminClient;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+
+/**
+ * Create an ApplicationScoped bean to cache AdminClient instances by connection ID and client ID.
+ */
+@ApplicationScoped
+public class AdminClients extends Clients<AdminClient> {
+
+  private static final String DEFAULT_CLIENT_ID = "default";
+  // Evict cached AdminClient client after 5 minutes of inactivity
+  private static final String CAFFEINE_SPEC = "expireAfterAccess=5m";
+
+  @Inject
+  ClusterCache cache;
+
+  @ConfigProperty(name = "ide-sidecar.admin-client-defaults")
+  Map<String, String> adminClientDefaults;
+
+  /**
+   * Get an AdminClient for the given connection ID and Kafka cluster ID.
+   * If the client does not already exist, it will be created.
+   * If the cluster ID is null, the default client ID will be used.
+   * @param connectionId The connection ID
+   * @param clusterId The cluster ID
+   * @return The AdminClient
+   */
+  public AdminClient getAdminClient(String connectionId, @Nullable String clusterId) {
+    if (clusterId != null) {
+      return getClient(
+          connectionId,
+          clusterId,
+          () -> AdminClient.create(getAdminClientConfig(connectionId, clusterId))
+      );
+    } else {
+      return getClient(
+          connectionId,
+          DEFAULT_CLIENT_ID,
+          () -> AdminClient.create(getAdminClientConfig(connectionId))
+      );
+    }
+  }
+
+  public Properties getAdminClientConfig(String connectionId, String clusterId) {
+    var props = new Properties();
+    var cluster = cache.getKafkaCluster(connectionId, clusterId);
+    props.put("bootstrap.servers", cluster.bootstrapServers());
+    props.putAll(adminClientDefaults);
+    return props;
+  }
+
+  /**
+   * Get AdminClient from the first available Kafka cluster for the connection. We implicitly assume
+   * that there will be only one Kafka cluster per connection.
+   * @param connectionId The connection ID
+   * @return The AdminClient configuration
+   */
+  public Properties getAdminClientConfig(String connectionId) {
+    var cluster = cache
+        .forConnection(connectionId)
+        // Accessing a ConcurrentHashMap _may_ be a blocking operation
+        .kafkaClusters
+        .values()
+        .stream()
+        .findFirst()
+        .orElseThrow(
+            () -> new AdminClientInstantiationException(
+                "No Kafka clusters found for connection " + connectionId
+            )
```

Comment:
Do we want to throw ClusterNotFoundException here, which is a bit more accurate? Or do we need this `AdminClientInstantiationException` type in particular?

Reply:
I think we need the `AdminClientInstantiationException` specifically here because we will only get to this code path when listing clusters (`GET /internal/kafka/v3/clusters`) on a connection that has no Kafka clusters associated, leading to a `400` status code (open to making this more accurate) -- see [exception mapper](https://github.com/confluentinc/ide-sidecar/blob/7c04ed8469dce896fe5d75f38ab94ce852370d57/src/main/java/io/confluent/idesidecar/restapi/exceptions/ExceptionMappers.java#L228-L240). However, using `ClusterNotFoundException` results in a 404 which feels inaccurate.

Considering that the internal endpoints are NOT meant to be _directly_ invoked by clients of ide-sidear, I figured simply handling the error scenario and bubbling up mostly appropriate error status codes/messages is sufficient.

Reply:
Going to let this be `AdminClientInstantiationException` unless you think otherwise, given my reasoning above.

Reply:
> I think we need the AdminClientInstantiationException specifically here because we will only get to this code path when listing clusters (GET /internal/kafka/v3/clusters) on a connection that has no Kafka clusters associated, leading to a 400 status code (open to making this more accurate) -- see [exception mapper](https://github.com/confluentinc/ide-sidecar/blob/7c04ed8469dce896fe5d75f38ab94ce852370d57/src/main/java/io/confluent/idesidecar/restapi/exceptions/ExceptionMappers.java#L228-L240). However, using ClusterNotFoundException results in a 404 which feels inaccurate.

Hmm, seems like listing clusters (GET /internal/kafka/v3/clusters) on a connection that has no Kafka clusters associated would return an empty list, not fail. No?

Reply:
You.. would be right, thanks! I didn't see it that way initially, but makes total sense.